### ISSUE TITLE
Youtube Plugin works but very high load times

### ISSUE THREAD

**Issue by MisterT87** [jdf76/plugin.video.youtube#722]
Hello Folks and Thanks for your great work on this addon.

I am using Kodi 18.5 and YOutube Plugin 6.5.2 on my linux machine.
I can load/start the addon without any errors and also the rest works (in general).

But every (except settings window) acction takes around 5-6 minutes (Opening subscriptions, youtube login etc).
Also Youtube login within the addon worked w/o problems, but took 15 minutes (every window which display the google code took also 6 minutes to load).

I didnt test video playback because it took to much time.

Some informations:
Kodi Logfile: [pastebin.com](https://pastebin.com/DeE3ZdzW)

System info:
Intel(R) Pentium(R) CPU N4200 @ 1.10GHz, 1805 MHz (quad core)
4 GB RAM
Intel(R) HD Graphics 505 (Broxton)

Linux: Manjaro deepin x64 (Arch based)

Kodi Info:
Kodi 18.5
Youtube Addon Version: 6.5.2
GPU acceleration enabled (VAAPI)

Hope someone can help and I didnt forget anything :D
Have a nice day!

**Comment by anxdpanic** [comment]
No clear reason I can see from the logs. Other users with reporting abnormally slow found the cause to be ipv6 connection. If that applies to you, disabling ipv6 for Kodi is done by https://forum.kodi.tv/showthread.php?tid=251033&pid=2748934#pid2748934

**Comment by MisterT87** [comment]
Whoa nice man for this big hint, Id never recognized these - But I cant test it today so please let this issue oped so I will report...

do you know what kodi is preffered to use ? has ipv6 higher priorty or ipv4 ? 

I am asking coz I am using both ipv4 / v6 same time (dual stack) and also have complex network setup lets say ^^

**Comment by anxdpanic** [comment]
Not sure, but I believe it prefers v6 when available. Which would be why disabling v6 in Kodi resolved the issue for the users. Note this was v6 between user and youtube being slow, not a v6 issue with Kodi.

**Comment by MisterT87** [comment]
So you think its not an python issue and wrong interpreting of ipv6 stuff or else and more of an issue of slow ipv6 (youtube) hosts ?

**Comment by anxdpanic** [comment]
In the past cases it's been the v6 connection to youtube, yes.

**Comment by MisterT87** [comment]
thanks for that hint also :D
first I will test it with v6 disabled completly in kodi - if solved then I reenable v6 again (need v6 in kodi) and using netfilter to solve this (TCP manipulation)....

**Comment by MisterT87** [comment]
TL DR: Problems are gone and solved and didnt changed anything - neither disabled ipv6!

Just booted up my machine and started kodi and only one kodi addon popped up: TwitchTV (Source: [Github](https://github.com/MrSprigster/Twitch-on-Kodi) .

started youtube addon and anything is fine (load times now around 3-5 seconds per page). Video playback also working fine now...

I didnt trace it down, but I think the previous version of TwitchTV something was really bad and fucked up my python or so...
I had one problem with another addon too, also gone now :)
thanks again!

Note: Let Issue open for answers maybe

**Comment by anxdpanic** [comment]
Thanks for the follow up, I'm going to close the issue. It will still be available via search and comments/re-open still available.
